src/Pact/Analyze/Eval/Numerical.hs

```diff
@@ -187,3 +191,33 @@ evalRoundingLikeOp2 op xT precisionT = do
 -- Round a real exactly between two integers (_.5) to the nearest even
 banker'sMethodS :: S Decimal -> S Integer
 banker'sMethodS (S prov x) = S prov $ banker'sMethod x
+
+evalBitwiseOp
+  :: Analyzer m
+  => BitwiseOp
+  -> [TermOf m 'TyInteger]
+  -> m (S Integer)
+evalBitwiseOp BitwiseAnd [xT, yT] = do
+  S _ x <- eval xT
+  S _ y <- eval yT
+  pure $ sansProv $ x .&. y
+evalBitwiseOp BitwiseOr [xT, yT] = do
+  S _ x <- eval xT
+  S _ y <- eval yT
+  pure $ sansProv $ x .|. y
+evalBitwiseOp Xor [xT, yT] = do
+  S _ x <- eval xT
+  S _ y <- eval yT
+  pure $ sansProv $ x `xor` y
+evalBitwiseOp Complement [xT] = do
+  S _ x <- eval xT
+  pure $ sansProv $ complement x
+evalBitwiseOp Shift [xT, yT] = do
+  S _ x <- eval xT
+  S _ y <- eval yT
+  case unliteral y of
```

Comment:
Note that [`shift`](http://hackage.haskell.org/package/base-4.12.0.0/docs/Data-Bits.html#t:Bits) from sbv has a different signature from the other bitwise ops -- its second argument is a concrete `Int`. In cases like this, we require a statically determined argument via `unliteral`. This means that it would be possible for the user to analyze a contract that calls `(shift x 10)` or even `(shift x (+ 5 5))`, but not `(shift x y)` for arbitrary `y`.